common/src/main/java/com/wynntils/core/discord/DiscordManager.java

```diff
@@ -0,0 +1,92 @@
+/*
+ * Copyright © Wynntils 2023.
+ * This file is released under AGPLv3. See LICENSE for full license details.
+ */
+package com.wynntils.core.discord;
+
+import com.wynntils.antiope.core.DiscordGameSDKCore;
+import com.wynntils.antiope.core.type.CreateParams;
+import com.wynntils.antiope.manager.activity.type.Activity;
+import com.wynntils.core.WynntilsMod;
+import com.wynntils.core.components.Manager;
+import com.wynntils.core.text.PartStyle;
+import com.wynntils.core.text.StyledText;
+import com.wynntils.mc.event.TickAlwaysEvent;
+import com.wynntils.models.character.type.ClassType;
+import com.wynntils.utils.mc.McUtils;
+import java.time.Instant;
+import java.util.List;
+import java.util.Locale;
+import net.minecraftforge.eventbus.api.SubscribeEvent;
+
+public class DiscordManager extends Manager {
+
+    private DiscordGameSDKCore core;
+    private Activity activity;
+
+    private int level = 0;
+    private ClassType classType = null;
+
+    public DiscordManager() {
+        super(List.of());
+
+        DiscordGameSDKCore.loadLibrary();
+        CreateParams params = new CreateParams();
+        try {
+            params.setClientID(387266678607577088L);
+            params.setFlags(CreateParams.getDefaultFlags());
+            core = new DiscordGameSDKCore(params);
+            activity = new Activity();
+            activity.timestamps().setStart(Instant.now());
+        } catch (RuntimeException e) {
+            WynntilsMod.error("Could not initialize Discord Game SDK", e);
+        }
```

Comment:
I don't think this should run here. Let's add some dependency tree in here, basically like `ChatHandler`'s `dialogExtractionDependents`. I want to give the user the ability to completely bypass running Discord SDK. With this said, you need to move this to an enable method, and create a disable method where you need to clean up all your `AutoCloseable`s.

Lastly, I think you should widen the catch to `Throwable` however I don't think we can catch SDK loading errors.